src/lib/PDF.cpp

```diff
@@ -191,93 +280,107 @@ pair<double, double> PDF::uncMin(double dchi) {
 }
 
 /*
- Derive the mode of the PDF, as well as the uncertainties
- Deal with asymetric PDF for the uncertainties
+ Derive the credible interval of the PDF, possibly asymetric
  First argument is the confidence level to consider, and the second argument is
- the considered value
+ the considered value centering the interval (typically the mode in the redshift
+ distribution)
  */
 pair<double, double> PDF::credible_interval(float level, double val) {
   pair<double, double> result;
   vector<double>::iterator bound_val, bound_left_id, bound_right_id, boundpos,
       boundneg;
 
+  // val is the centered value of the credible interval; it normally has to be
+  // located inside the xaxis. But we have pdfmaps that will never be correct,
+  // e.g. when BC03 is not used (no physical parameter estimation), and thus we
+  // return an empty interval.
+  // Likewise when level <= 0, which is not a meaningful value.
+  if (val < scaleMin || val >= scaleMax || level <= 0.) {
+    return make_pair(val, val);
+  }
+
   // if levels given in percentage
   if (level > 1.) level /= 100.;
-
-  // Find the index corrresponding to the value given in input
-  bound_val = upper_bound(xaxis.begin(), xaxis.end(), val);
-  size_t maxid = bound_val - xaxis.begin();
-  // Take the index of the closest value
-  if (maxid > 0) {
-    if ((xaxis[maxid] - val) > (val - xaxis[maxid - 1])) maxid = maxid - 1;
-  }
+  // we assume that level>1 means level provided as a percentage
+  // but this means that now level should be <1
+  if (level > 1.) return make_pair(scaleMin, scaleMax);
 
   // Compute the full cumulant
   vector<double> cumulant;
   double tmp = 0;
   cumulant.push_back(0.0);
-  for (size_t k = 0; k < xaxis.size() - 1; k++) {
+  for (size_t k = 0; k < vsize - 1; k++) {
```

Comment:
Is there any reason to sometimes use xaxis.size() and sometimes use vsize. Not important but was this changed for a reason and should we be consistent?